ethexe/observer/src/sync.rs

```diff
@@ -40,17 +40,17 @@ use std::{
 };
 
 // TODO (gsobol): make tests for ChainSync
-pub(crate) struct ChainSync {
+pub(crate) struct ChainSync<DB: OnChainStorage + CodesStorage> {
```

Comment:
if im not mistaken we were discussion about this, what dont we just put Database itself?

Reply:
Here we have no problems to define as generic because:
1) it's not public
2) here we specify what chain sync could do : just fill codes and on-chain storages

Reply:
Discussed - ok